consul/tests/test_unit.py

```diff
@@ -117,7 +117,7 @@ def test_get_nodes_with_service_critical(aggregator):
 
 def test_consul_request(aggregator, instance):
     consul_check = ConsulCheck(common.CHECK_NAME, {}, {})
-    with mock.patch("datadog_checks.consul.consul.requests") as mock_requests:
+    with mock.patch("datadog_checks.consul.consul.requests.get") as mock_requests_get:
```

Comment:
mocking the whole module makes `requests.exceptions.Timeout` a class of type `mock` instead of a subclass of `BaseException`. In python 3, trying to catch something that is not an exception in https://github.com/DataDog/integrations-core/blob/dea1f9439c589200bb8291080b3d5c21ece95a3a/consul/datadog_checks/consul/consul.py#L110 raises `TypeError: catching classes that do not inherit from BaseException is not allowed`.